crates/oxc_linter/src/external_plugin_store.rs

```diff
@@ -119,6 +129,27 @@ impl ExternalPluginStore {
         let plugin = &self.plugins[external_rule.plugin_id];
         (&plugin.name, &external_rule.name)
     }
+
+    /// Add options to the store and return its ID.
+    /// Returns index 0 for empty arrays or null values (no options).
+    pub fn add_options(&mut self, options: serde_json::Value) -> ExternalOptionsId {
+        // If it's null or an empty array, return reserved index 0
+        if options.is_null() || options.as_array().is_some_and(Vec::is_empty) {
+            return ExternalOptionsId::from_usize(0);
+        }
+
+        self.options.push(options)
+    }
```

Comment:
As far as I can see, `ESLintRule::config` field is always `None` or it's `Some` containing `serde_json::Array` with not-zero length. It's never `Some` containing any other type.

https://github.com/oxc-project/oxc/blob/43a6c3209e7c03e3b83d37879ddf7ff919f44d2b/crates/oxc_linter/src/config/rules.rs#L269-L306

This is poor design. `ESLintRule::config` should be `Option<Vec<serde_json::Value>>`, not `Option<serde_json::Value>`. But that's another story...

For now, here could just do:

```rs
let options = options.as_array().unwrap();
debug_assert!(!options.is_empty(), "`options` should never be an empty `Vec`");
self.options.push(options)
```